packages/connectivity/connectivity_macos/macos/Classes/ConnectivityPlugin.swift

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Cocoa
+import CoreWLAN
+import FlutterMacOS
+import Reachability
+import SystemConfiguration.CaptiveNetwork
+
+public class ConnectivityPlugin: NSObject, FlutterPlugin, FlutterStreamHandler {
+  var reach: Reachability?
+  var eventSink: FlutterEventSink?
+  var cwinterface: CWInterface?
+
+  public override init() {
+    cwinterface = CWWiFiClient.shared().interface()
+  }
+
+  public static func register(with registrar: FlutterPluginRegistrar) {
+    let channel = FlutterMethodChannel(
+      name: "plugins.flutter.io/connectivity",
+      binaryMessenger: registrar.messenger)
+
+    let streamChannel = FlutterEventChannel(
+      name: "plugins.flutter.io/connectivity_status",
+      binaryMessenger: registrar.messenger)
+
+    let instance = ConnectivityPlugin()
+    streamChannel.setStreamHandler(instance)
+
+    registrar.addMethodCallDelegate(instance, channel: channel)
+  }
+
+  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+    switch call.method {
+    case "check":
+      result(statusFromReachability(reachability: Reachability.forInternetConnection()))
+    case "wifiName":
+      result(cwinterface?.ssid())
+    case "wifiBSSID":
+      result(cwinterface?.bssid())
+    case "wifiIPAddress":
+      result(getWifiIP())
+    default:
+      result(FlutterMethodNotImplemented)
+    }
+  }
+
+  /// Returns a string describing connection type
+  ///
+  /// - Parameters:
+  ///   - reachability: an instance of reachability
+  /// - Returns: connection type string
+  private func statusFromReachability(reachability: Reachability?) -> String {
+    // checks any non-WWAN connection
+    if reachability?.isReachableViaWiFi() ?? false {
```

Comment:
What is the `?? false` accomplishing here? (I'm not very familiar with Swift's syntax. I'm guessing it's saying false if anything in the previous statement evaluates to null, but if that's the case my guess would be that it's not needed since the null aware operators would evaluate to false anyway. Swift may be more picky about types than that though.)

Reply:
You are mostly correct on how that works except on the optional object part :). The optional has to be unwrapped or it will raise  a compiler error. With this syntax we don't have to unwrap it, but just supply a default false value. It's less verbose than unwrapping it above and still supply the default value.

Reply:
Got it, thanks!